Detectors/ITSMFT/MFT/base/src/Barrel.cxx

```diff
@@ -0,0 +1,430 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "TGeoManager.h"
+#include "TGeoMatrix.h"
+#include "TGeoManager.h"
+#include "TGeoCompositeShape.h"
+#include "TGeoShape.h"
+#include "TGeoCone.h"
+#include "TGeoVolume.h"
+#include "TGeoMaterial.h"
+#include "TGeoMedium.h"
+#include "TGeoTube.h"
+#include "TGeoTrd1.h"
+#include "TMath.h"
+#include "TGeoXtru.h"
+
+#include "MFTBase/Barrel.h"
+
+using namespace o2::MFT;
+
+ClassImp(o2::MFT::Barrel);
+
+//_____________________________________________________________________________
+Barrel::Barrel() : TNamed(), mBarrel(nullptr)
+{
+
+  // default constructor
+}
+
+//_____________________________________________________________________________
+Barrel::~Barrel() = default;
+
+//_____________________________________________________________________________
+TGeoVolumeAssembly* Barrel::createBarrel(Int_t half)
```

Comment:
half argument is provided but never used

Reply:
The exact positioning of this new volume is not perfectly defined. Then we don't create the volume, it will be perform later. If it's required anyway, I would propose to add its creation in GeometryBuilder under the condition of a flag (by default, no creation).

Reply:
@frmanso up to you if call it now or later. What about the argument, looks like copy-paste leftover.